generator/src/main/java/com/graphicsfuzz/generator/tool/Generate.java

```diff
@@ -202,7 +212,33 @@ public static StringBuilder generateVariant(ShaderJob shaderJob,
     }
 
     if (args.getGenerateUniformBindings()) {
-      shaderJob.makeUniformBindings();
+      Optional<String> pushConstant = Optional.empty();
+      if (random.nextFloat() <= args.getPushConstantProbability()) {
+        // Get the list of (unique) uniform names in all shaders
+        final Set<String> allUniforms = new HashSet<String>();
+        for (TranslationUnit tu : shaders) {
+          if (tu.getShadingLanguageVersion().supportedPushConstants()) {
+            for (Declaration decl : tu.getTopLevelDeclarations()) {
+              if (decl instanceof VariablesDeclaration
+                  && ((VariablesDeclaration) decl).getBaseType()
+                  .hasQualifier(TypeQualifier.UNIFORM)) {
```

Comment:
I think we should also check that the base type is a BasicType.  For example sampler types probably cannot be push constants.

Reply:
Done